sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -130,6 +130,34 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Buckets the output by the given columns on the file system. If specified, the output is
+   * laid out on the file system similar to Hive's bucketing scheme.
+   *
+   * This is applicable for Parquet, JSON and ORC.
+   *
+   * @since 2.0
+   */
+  @scala.annotation.varargs
+  def bucketBy(numBuckets: Int, colName: String, colNames: String*): DataFrameWriter = {
+    this.numBuckets = Option(numBuckets)
+    this.bucketColumnNames = Option(colName +: colNames)
+    this
+  }
+
+  /**
+   * Sorts the bucketed output by the given columns.
+   *
+   * This is applicable for Parquet, JSON and ORC.
```

Comment:
Why the limitation? The comment makes this seem bucketBy must be called first. If that is true, make that more clear.